src/pgduckdb_duckdb.cpp

```diff
@@ -167,17 +161,19 @@ DuckDBManager::LoadSecrets(duckdb::ClientContext &context) {
 		DuckDBQueryOrThrow(context, secret_key->data);
 
 		pfree(secret_key->data);
-		secret_id++;
-		secret_table_num_rows = secret_id;
+		++secret_id;
 	}
+
+	secret_table_num_rows = secret_id;
```

Comment:
I think this was better in the previous place. That way even if `LoadSecrets` fails midway through when adding e.g. the 3rd secret, the 2 previously created secrets would be cleaned up by `DropSecrets`. Ofcourse we should probably handle this with transactions, but we currently don't. So it seems better to keep this line in the same place, and probably add a comment that it makes sense to have it there for now.

Reply:
Yeah - I actually wanted to make it so if one fails it doesn't arbitrarily fails all the following ones but didn't finish. I might do that in a different PR, but will revert in the meantime.